### PR TITLE
Added server connection status check

### LINKED ISSUE #23: Server status display
Add a small server status icon in the index.html, something like : 
green - Server online
orange - Connecting (default)
red - Server connection failed

Rough implementation idea:
Send an API request to the server, if the server returns True, display green, if no response display red.

### PR DESCRIPTION
**Pull request by farhadjaman**
### Details and comments
closes #23

### CODE REVIEW

File: report_ai/index.html [hunk 1]
```diff
@@ -35,7 +35,7 @@
     <div style = "width:75%"><h1>Report.AI</h1></div>
     <div style = "display:flex; flex-direction:column; margin-left:7%">
       <div style="height:70%;"><button title="Reload Current-tab" id = "refreshBtn">Reload</button></div>
-      <div class = "connectionSignal"> </div>
+      <div class = "connectionSignal" id="connectionSignal"> </div>
     </div>
```
Comment (AnsahMohammad) [inline report_ai/index.html:39]: Can we add a title="waiting for connection"
and in the script, 
if server alive, set title = "connected to server"
else title = "server unavailable"


Everything else looks good :fire:

**Review by AnsahMohammad — CHANGES_REQUESTED** [review]

File: report_ai/index.html (hunk 1 above)
Comment (farhadjaman) [reply report_ai/index.html:39]: yeah, i will make the changes. Thanks for the review

Comment (farhadjaman) [reply report_ai/index.html:40]: Enhance connection signal element with text and icon integration. Review requested.

**Comment by AnsahMohammad** [comment]
Looking Good !, that's a huge UX enhancement

**Comment by farhadjaman** [comment]
Thank you!

**Comment by gobbledy-gook** [comment]
Appreciate it @farhadjaman ! Good work !

**Comment by farhadjaman** [comment]
Thank you @gobbledy-gook !

**Pull request merged**